packages/otelbin/src/components/monaco-editor/parseYaml.ts

```diff
@@ -98,16 +99,17 @@ export const getYamlDocument = (editorValue: string) => {
 	return docElements;
 };
 
-export function parseYaml(yamlItems: IItem[]) {
+export function parseYaml(yamlItems: IItem[], path: string[] = []) {
 	const parsedYamlConfig: IYamlElement[] = [];
 	if (!yamlItems) return;
 	else if (Array.isArray(yamlItems)) {
 		for (const item of yamlItems) {
 			if (item) {
 				const key = item.key?.source ?? item.value?.source;
 				const keyOffset = item.key?.offset ?? item.value?.offset;
-				const value = parseYaml(item.value?.items) ?? item.value?.source;
-				parsedYamlConfig.push({ key: key, offset: keyOffset, value: value });
+				const newPath = path.concat([key]);
+				const value = parseYaml(item.value?.items, newPath) ?? item.value?.source;
+				parsedYamlConfig.push({ key: key, offset: keyOffset, value: value, path: newPath });
```

Comment:
**Question:** What is `path` being used for?

Reply:
@bripkens I added it for the initial proposal (separating serverSide and browserSide) with comparing error path. and didn't remove it, I thought it will be useful for the future features.

Reply:
Let's not keep dead code around. If we need it down the road, we can always restore it from PRs we haven't merged/code we have changed :)